### DISCUSSION TITLE
Per character text colour on GTK TextArea?

### DISCUSSION THREAD

**Discussion by Beyley** [picoe/Eto#2177] [Q&A] [answered]
I am trying to colour the text in a text area more dynmically than just the whole text area, and it seems like theres no way to do that under Linux, as the GTK backend doesnt support RTF, and none of the other backends work on linux, is there no way to do this?

**Comment by cwensley** [comment, marked as answer]
While GTK doesn't support RTF, you should still be able to use [RichTextArea](https://github.com/picoe/Eto/blob/develop/src/Eto/Forms/Controls/RichTextArea.cs#L168) to add colour, font, and size to various spans in the control.

**Reply by Beyley** [reply]
> While GTK doesn't support RTF, you should still be able to use [RichTextArea](https://github.com/picoe/Eto/blob/develop/src/Eto/Forms/Controls/RichTextArea.cs#L168) to add colour, font, and size to various spans in the control.

it seems i have been trolled by my computer, when i tried to set the forground colour an hour ago through that, it would just crash with an infinitely printing log that i assumed was just some weird native crash cus no RTF, but *now* when i try, it just works for some reason, so i guess i can close this
